siege_utilities/geo/boundary_providers.py

```
@@ -227,3 +228,141 @@ def resolve_boundary_provider(country: str = 'US', **kwargs: Any) -> BoundaryPro
+
+    def list_levels(self) -> list[str]:
+        """Return the geographic levels this provider supports."""
+        return list(self._LEVELS)
+
+    def is_available(self) -> bool:
+        """Return True if geopandas is installed and credentials are set."""
+        if not self._username:
+            return False
+        try:
+            import geopandas  # noqa: F401
+            return True
```

Ruff (0.15.10)

[warning] Consider moving this statement to an `else` block

(TRY300)